services/task-svc/internal/tracking/tracking.go

```diff
@@ -0,0 +1,51 @@
+package tracking
+
+import (
+	"common"
+	"context"
+	"decaying_lru"
+	zlog "github.com/rs/zerolog/log"
+	"time"
+)
+
+const (
+	PatientKey = "recently-viewed-patient"
+)
+
+var lru decaying_lru.DecayingLRU
+
+func SetupTracking(serviceName string) {
+	lru = decaying_lru.Setup(serviceName, 5, 4*24*time.Hour, 10)
+}
+
+func getUserID(ctx context.Context) string {
+	log := zlog.Ctx(ctx)
+
+	userID, err := common.GetUserID(ctx)
+
+	if err != nil {
+		log.Error().Err(err).Msg("could not get userID from context, can't add item to LRU")
+		return ""
+	}
```

Comment:
This error handling make more sense inside the `AddPatientToRecentActivity` function.

Reply:
No, it will be required in `AddPatientToRecentActivity`, `RemovePatientFromRecentActivity`, `AddWardToRecentActivity` and `RemoveWardFromRecentActivity`, possibly even more. I don't want to repeat this everywhere